### PR TITLE
fix(modal container): support for custom styles

### PR DESCRIPTION
**Pull request by arjunlalb**
## Description
Support applying custom styles in modal container

### Testing
Tested manually. 

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: projects/components/src/modal/modal.ts [hunk 1]
```diff
@@ -8,6 +9,7 @@ export interface ModalConfig<TData = unknown> {
   title?: string;
   data?: TData;
   closeOnEscapeKey?: boolean;
+  styles?: Dictionary<unknown>;
```
Comment (anandtiwary) [inline projects/components/src/modal/modal.ts:12]: allowing arbitrary styles here could be bad. Can we just support only those customization that we need? Let's quickly discuss this on monday.

Comment (arjunlalb) [reply projects/components/src/modal/modal.ts]: @anandtiwary fixed it by defining an interface for style.

File: projects/components/src/modal/modal.ts [hunk 2]
```diff
@@ -9,7 +9,11 @@ export interface ModalConfig<TData = unknown> {
   title?: string;
   data?: TData;
   closeOnEscapeKey?: boolean;
-  styles?: Dictionary<unknown>;
+  styles?: ModalStyles;
+}
+
+export interface ModalStyles {
```
Comment (anandtiwary) [inline projects/components/src/modal/modal.ts:15]: I still have reservations about this. Could you please add a todo here? We should come back and discuss this before adding more styles

**Review by anandtiwary — APPROVED** [review]

File: projects/components/src/modal/modal.ts (hunk 2 above)
Comment (anandtiwary) [reply projects/components/src/modal/modal.ts:15]: approving for now.

Comment (arjunlalb) [reply projects/components/src/modal/modal.ts:16]: @anandtiwary added a todo. please reapprove :)

**Review by anandtiwary — APPROVED** [review]

**Pull request merged**